File: kata-8-kyu/convert-boolean-values-to-strings-yes-or-no/python/convert_boolean_values_to_strings_yes_or_no_test.py
# + =====================================================================================================================
# +
# + 8 kyu - Convert boolean values to strings 'Yes' or 'No'.  [ ID: 53369039d7ab3ac506000467 ] (convert-boolean-values-to-strings-yes-or-no)
# + URL: https://www.codewars.com/kata/53369039d7ab3ac506000467
# + Category: REFERENCE  |  Tags: FUNDAMENTALS
# +
# + =====================================================================================================================

import codewars_test as test
from convert_boolean_values_to_strings_yes_or_no import bool_to_word


@test.describe("Fixed Tests")
def basic_tests():
    @test.it("Basic Test Cases")
    def basic_test_cases():
        test.assert_equals(bool_to_word(True), "Yes")
        test.assert_equals(bool_to_word(False), "No")


@test.describe("Random Tests")
def random_tests():

    from random import choice

    sol = lambda b: ["No", "Yes"][b]

    for _ in range(100):
        bool = choice([True, False])

        @test.it(f"Testing for bool_to_word({bool})")
        def test_case():
            test.assert_equals(bool_to_word(bool), sol(bool))

File: kata-8-kyu/convert-boolean-values-to-strings-yes-or-no/javascript/convertBooleanValuesToStringsYesOrNo.Test.js
//+ ====================================================================================================================
//+
//+ 8 kyu - Convert boolean values to strings 'Yes' or 'No'.  [ ID: 53369039d7ab3ac506000467 ] (convert-boolean-values-to-strings-yes-or-no)
//+ URL: https://www.codewars.com/kata/53369039d7ab3ac506000467
//+ Category: REFERENCE  |  Tags: FUNDAMENTALS
//+
//+ ====================================================================================================================

import { assert } from "vitest"
const { boolToWord } = require("./convertBooleanValuesToStringsYesOrNo")

describe("Basic tests", () => {
  it("Testing for basic tests", () => {
    assert.strictEqual(boolToWord(true), "Yes")
    assert.strictEqual(boolToWord(false), "No")
  })

  it("Testing for true values", () => {
    assert.strictEqual(boolToWord(0 < 1), "Yes")
    assert.strictEqual(boolToWord(!(0 > 1)), "Yes")
    assert.strictEqual(boolToWord(typeof boolToWord == "function"), "Yes")
    assert.strictEqual(boolToWord(!0), "Yes")
  })

  it("Testing for false values", () => {
    assert.strictEqual(boolToWord(0 > 1), "No")
    assert.strictEqual(boolToWord(!(0 < 1)), "No")
    assert.strictEqual(boolToWord(typeof boolToWord == "string"), "No")
    assert.strictEqual(boolToWord(!1), "No")
  })
})

describe("Random tests", () => {
  const arr = [true, false]

  function randint(min, max) {
    return Math.floor(Math.random() * (max - min + 1)) + min
  }

  for (let i = 0; i < 100; i++) {
    let b = arr[randint(0, 1)]
    let expected = b ? "Yes" : "No"
    it(`boolToWord(${b}) should equal ${JSON.stringify(expected)}`, () => {
      assert.strictEqual(boolToWord(b), expected)
    })
  }
})
